### DISCUSSION TITLE
Receive note not from USB, but from 5DIN

### DISCUSSION THREAD

**Discussion by santta** [tttapa/Control-Surface#868] [Q&A]
Hi, how to receive midi note on arduino, not from usb, but from 5din?

I'm using a sketch from the exsample library that lights up a diode when a note (C4) is received via USB. And I need to receive note, from port 5din.

        
        #include <Control_Surface.h> // Include the Control Surface library
        
        // Instantiate a MIDI over USB interface.
        USBMIDI_Interface midi;
        
        using namespace MIDI_Notes;
        
        // Instantiate the LED that will light up when middle C is playing
        NoteValueLED led = {
          LED_BUILTIN,             // Pin of built-in LED
          {note(C, 4), CHANNEL_1}, // Note C4 on MIDI channel 1
        };
        
        void setup() {
          Control_Surface.begin(); // Initialize Control Surface
        }
        
        void loop() {
          Control_Surface.loop(); // Update the Control Surface
        }

**Comment by tttapa** [comment]
See https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/df7/midi-tutorial.html#midi_md-midi-interfaces, you can use [`HardwareSerialMIDI_Interface`](https://tttapa.github.io/Control-Surface-doc/Doxygen/d9/d78/classHardwareSerialMIDI__Interface.html).
